crates/notion-core/src/distro/mod.rs

```diff
@@ -37,47 +38,26 @@ impl<V> Fetched<V> {
     }
 }
 
-/// Abstraction to contain info about Distro versions.
-#[derive(Eq, PartialEq, Clone, Debug)]
-pub enum DistroVersion {
-    // the version of the Node runtime, and the npm version installed with that
-    Node(Version, Version),
-    Yarn(Version),
-    Npm(Version),
-    Package(String, Version),
-}
-
-impl Display for DistroVersion {
-    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), fmt::Error> {
-        let s = match self {
-            &DistroVersion::Node(ref runtime, ref npm) => {
-                format!("node version {} (npm {})", runtime, npm)
-            }
-            &DistroVersion::Yarn(ref version) => format!("yarn version {}", version),
-            &DistroVersion::Npm(ref version) => format!("npm version {}", version),
-            &DistroVersion::Package(ref name, ref version) => {
-                format!("{} version {}", name, version)
-            }
-        };
-        f.write_str(&s)
-    }
-}
-
 pub trait Distro: Sized {
     type VersionDetails;
+    type ResolvedVersion;
 
-    /// Provisions a new Distro based on the Version and Possible Hooks
-    fn new(version: Version, hooks: Option<&ToolHooks<Self>>) -> Fallible<Self>;
+    /// Provisions a new Distro based on the name, Version and Possible Hooks
+    fn new(
+        name: String,
```

Comment:
I like that idea - I'll pull those `new()` constructors out of `Distro`.

Reply:
Actually, looking at this now, the `new()` constructor is needed for the `resolve()` method of `FetchResolve`. I don't think that can be removed from `Distro`, at least not in an obvious way.

Reply:
Ah, that makes sense. No worries on this then, it can probably be re-evaluated along with the potential investigation into whether it's possible to combine the `fetch` methods again (well outside the scope of this PR, though).